src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs

```diff
@@ -96,9 +101,39 @@ private void CloseButton_Click(object sender, RoutedEventArgs e)
             viewModel.PackagePathsViewModel.SaveSettingCommand.Execute(null);
             viewModel.CommitPackagePathsForInstall();
             PackagePathView.Dispose();
+
+            RunGraphWhenScaleFactorUpdated();
+
             Close();
         }
 
+        /// <summary>
+        /// This method will run the graph only if the Geometry Scaling was updated otherwise will not be executed
+        /// </summary>
+        private void RunGraphWhenScaleFactorUpdated()
+        {
+            //If the new radio button selected (ScaleValue) is different than the current one in Dynamo, we update the current one
+            if (dynViewModel.ScaleFactorLog != scaleValue)
+            {
+                dynViewModel.ScaleFactorLog = scaleValue;
+                dynViewModel.CurrentSpace.HasUnsavedChanges = true;
+
+                //Due that binding are done before the contructor of this class we need to execute the Log only if the viewModel was assigned previously
+                if (viewModel != null)
+                {
+                    Log(String.Format("Geometry working range changed to {0} ({1}, {2})",
+                    viewModel.ScaleRange.Item1, viewModel.ScaleRange.Item2, viewModel.ScaleRange.Item3));
+                    viewModel.UpdateSavedChangesLabel();
+                    Dynamo.Logging.Analytics.TrackEvent(
+                        Actions.Switch,
+                        Categories.Preferences,
+                        Res.PreferencesViewVisualSettingsGeoScaling);
+                }
+
+                dynViewModel.ExecuteCommand(new DynamoModel.ForceRunCancelCommand(false, false));
```

Comment:
As discussed on slack please make the change back to:
```
var allNodes = dynViewModel.HomeSpace.Nodes;
                dynViewModel.HomeSpace.MarkNodesAsModifiedAndRequestRun(allNodes, forceExecute: true);
```
in place of `ForceRunCancelCommand`.

Reply:
> @RobertGlobant20 can you add tests simulating the UI setting changes?

@aparajit-pratap right now there is a UI Automation test validating that the Geometry Scaling functionality works (adds a node with a large number that sends a warning due that the Geometry Scaling is medium, then change the Geometry Scaling and the warning disappear).
then should I be adding a new UI Automation test that validates that if the RunMode = Manual and the Geometry Scaling is changed the graph is not executed and also if the the RunMode = Automatic and the Geometry Scaling is changed the graph is executed automatically, right?

Reply:
I think this test can be added in Dynamo as well, no?

Reply:
Ok, I will start the test implementation in Dynamo project.

Reply:
@aparajit-pratap I'm not sure if is possible to test this functionality using UnitTests in Dynamo project, because I was checking that the only way to execute the MarkNodesAsModifiedAndRequestRun() method (in RunMode Manual or Automatic) is by pressing the close button in the Preferences dialog and for doing that I will need to open the Preferences dialog (I not sure if this is possible using Unit Testing).
Also setting the Property (in ViewModel) that binds the Geometry Scaling Radio Buttons will not execute the Geometry_Scaling_Checked() method that sets the ScaleFactor variable unless you do it manually clicking the radio buttons.

Please let me know your ideas for knowing if this is possible otherwise I can create an UI Automation test for this behavior. Thanks